extensions/mssql/src/main.ts

```diff
@@ -150,6 +158,76 @@ function activateSparkFeatures(appContext: AppContext): void {
 	});
 }
 
+function activateNotebookTask(appContext: AppContext): void {
+	let apiWrapper = appContext.apiWrapper;
+	apiWrapper.registerTaskHandler(Constants.mssqlClusterNewNotebookTask, (profile: sqlops.IConnectionProfile) => {
+		return saveProfileAndCreateNotebook(profile);
+	});
+	apiWrapper.registerTaskHandler(Constants.mssqlClusterOpenNotebookTask, (profile: sqlops.IConnectionProfile) => {
+		return handleOpenNotebookTask(profile);
+	});
+}
+
+function saveProfileAndCreateNotebook(profile: sqlops.IConnectionProfile): Promise<void> {
+	return handleNewNotebookTask(undefined, profile);
+}
+
+async function handleNewNotebookTask(oeContext?: sqlops.ObjectExplorerContext, profile?: sqlops.IConnectionProfile): Promise<void> {
+	// Ensure we get a unique ID for the notebook. For now we're using a different prefix to the built-in untitled files
+	// to handle this. We should look into improving this in the future
+	let untitledUri = vscode.Uri.parse(`untitled:Notebook-${untitledCounter++}`);
+	let editor = await sqlops.nb.showNotebookDocument(untitledUri, {
+		connectionId: profile.id,
+		providerId: jupyterNotebookProviderId,
+		preview: false,
+		defaultKernel: {
+			name: 'pyspark3kernel',
+			display_name: 'PySpark3',
+			language: 'python'
+		}
+	});
+	if (oeContext && oeContext.nodeInfo && oeContext.nodeInfo.nodePath) {
```

Comment:
Minor: probably not needed but we'll handle merging correctly down the line when the code moves to ADS